File: templates/matlab/velocity_field_cfd.m
function fig = velocity_field_cfd()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    apply_theme();
    [X, Y] = meshgrid(linspace(-2,2,60), linspace(-1,1,40));
    U = 1 - X.^2 + 0.3*Y; V = -0.15*Y.*X;
    speed = hypot(U, V);
    fig = figure('Position',[100 100 700 500]);
    pcolor(X, Y, speed); shading interp; hold on;
    colormap(parula); colorbar;
    skip = 4;
    quiver(X(1:skip:end,1:skip:end), Y(1:skip:end,1:skip:end), ...
           U(1:skip:end,1:skip:end), V(1:skip:end,1:skip:end), 1.5, 'Color','w');
    xlabel('x'); ylabel('y'); title('Velocity field'); axis equal tight;
end

File: templates/python/velocity_field_cfd.py
"""velocity_field_cfd: 二维流场速度矢量 + 速度模值底色（CFD 风格）."""
import sys, os; sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
import numpy as np
import matplotlib.pyplot as plt
from theme import apply_theme

def make_figure(title='Velocity field'):
    apply_theme(fig_size=(7, 5))
    Y, X = np.mgrid[-1:1:40j, -2:2:60j]
    U = 1 - X**2 + 0.3*Y          # 类 Poiseuille
    V = -0.15*Y*X
    speed = np.hypot(U, V)
    fig, ax = plt.subplots()
    pc = ax.pcolormesh(X, Y, speed, cmap='viridis', shading='auto')
    skip = (slice(None, None, 4), slice(None, None, 4))
    ax.quiver(X[skip], Y[skip], U[skip], V[skip], color='white', scale=25)
    fig.colorbar(pc, ax=ax, label='|U|')
    ax.set_xlabel('x'); ax.set_ylabel('y'); ax.set_title(title); ax.set_aspect('equal')
    fig.tight_layout(); return fig

if __name__ == '__main__':
    fig = make_figure(); fig.savefig(__file__.replace('.py', '.png'), dpi=150)
